src/util/rocksdb.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rocksdb::{DB, DBCompressionType, Options};
+use rocksdb::rocksdb_ffi::DBCFHandle;
+
+pub fn get_compression_by_string(tp: &str) -> DBCompressionType {
+    match &*tp.to_owned().to_lowercase() {
+        "no" => DBCompressionType::DBNo,
+        "snappy" => DBCompressionType::DBSnappy,
+        "zlib" => DBCompressionType::DBZlib,
+        "bzip2" => DBCompressionType::DBBz2,
+        "lz4" => DBCompressionType::DBLz4,
+        "lz4hc" => DBCompressionType::DBLz4hc,
+        _ => panic!("unsupported compression type {}", tp),
+    }
+}
+
+pub fn get_cf_handle<'a>(db: &'a DB, cf: &str) -> Result<&'a DBCFHandle, String> {
+    db.cf_handle(cf)
+        .ok_or_else(|| format!("cf {} not found.", cf))
+}
+
+pub fn new_engine(path: &str, cfs: &[&str]) -> Result<DB, String> {
+    let opts = Options::new();
+    new_engine_opt(opts, path, cfs)
+}
+
+pub fn new_engine_opt(mut opts: Options, path: &str, cfs: &[&str]) -> Result<DB, String> {
+    // TODO: configurable opts for each CF.
+    // Currently we support 1) Create new db. 2) Open a db with CFs we want. 3) Open db with no
+    // CF.
+    // TODO: Support open db with incomplete CFs.
+    opts.create_if_missing(false);
```

Comment:
As I said, Rocksdb requires to be opened with all existed CFs, so the `open` at first line will fail.